### PR TITLE
[BD] Upgrade django-waffle 

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description 
This is a simple upgrade to 0.17.0, however in django-waffle changed the way how the override decorator were defined, They now are inheriting from `TestContextDecorator`(https://github.com/django-waffle/django-waffle/blob/master/waffle/testutils.py#L12), this raises the following error when we execute the tests.
`raise TypeError('Can only decorate subclasses of unittest.TestCase')`

that is way all the statements like `@override_switch...`, were relocated to the test class 

part of https://openedx.atlassian.net/browse/BOM-1360

### Reviewers
- [x] @morenol 
- [x]  Is this ready for edX's review? 
- [ ] @jmbowman 
 
### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @andrey-canon! I've created [OSPR-4197](https://openedx.atlassian.net/browse/OSPR-4197) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: requirements/constraints.txt [hunk 1]
```diff
@@ -34,6 +34,9 @@ django-crispy-forms<1.9.0
 # 4.0.0 drops Python 2.7 support
 django-model-utils<4.0.0
 
+# 0.18.0 drops Python 2.7 support
+django-waffle<0.18.0
```
Comment (morenol) [inline requirements/constraints.txt]: I think that the constraint should be django-waffle<0.19.0

https://github.com/django-waffle/django-waffle/blob/master/CHANGES

Comment (andrey-canon) [reply requirements/constraints.txt]: @morenol  yes you are right, addressed

**Review by morenol — APPROVED** [review]
LGTM, I would include this to the description of the PR https://github.com/django-waffle/django-waffle/pull/331

**Review by jmbowman — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@andrey-canon 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-analytics-dashboard&entry.867055334=https://github.com/edx/edx-analytics-dashboard/pull/956&entry.1484655318=https://github.com/andrey-canon&entry.752974735=2020-03-08+20:23&entry.1917517419=2020-03-09+13:54&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**